### PR TITLE
fixed unfolding active submenus on add/edit pages

### PR DESCRIPTION
**Pull request by Gooliver2**

### CODE REVIEW

**Comment by Gooliver2** [comment]
ok, I'll investigate how to fix more intelligence way.

**Comment by Gooliver2** [comment]
@mmarcos added new way of defining ARCTIC_MENU and folding side menu. 
Advantage of this way instead of using regex is that we can include into menu/submenu any urls, not only path-dependent. Also regex approach will probably fails if somebody will decide to put couple submenu rows into Home ('/') menu.

File: arctic/utils.py [hunk 1]
```diff
@@ -12,10 +12,25 @@
...
+            return True
+        elif len(menu_entry) >= 3:
+            related_urls = []
+            related_submenus = []
+
+            for menu_item in menu_entry[2:]:
+                if is_list_of_list(menu_item):
+                    # suppose it is submenus
+                    related_submenus = menu_item
+                elif type(menu_item) in (tuple, list):
```
Comment (wolendranh) [inline arctic/utils.py]: you can write this a bit shorter probably.
`elif isinstance(menu_item, (tuple, list))
`

File: arctic/utils.py [hunk 2]
```diff
@@ -322,3 +345,14 @@ def func_wrapper(self, start, stop):
...
         return func(self, offset, limit)
     return func_wrapper
+
+
+def is_list_of_list(item):
+    """
+    check whether the item is list (tuple)
+    and consist of list (tuple) elements
+    """
+    if type(item) in (list, tuple) and len(item) \
```
Comment (wolendranh) [inline arctic/utils.py]: same here about instance check:
`isinstance(item, (tuple, list)) and ...
`

File: arctic/utils.py [hunk 3]
```diff
@@ -12,10 +12,25 @@
 from . import defaults
 
 
-def is_active(path, path_to_check):
-    if not path_to_check:
-        return False
-    return path == path_to_check
+def is_active(menu_entry, url_name):
```
Comment (wolendranh) [inline arctic/utils.py]: Would be also nice to have some tests covering this logic.

**Comment by mmarcos** [comment]
Looks like a good solution!
Could you add some tests and perhaps throw an ImproperlyConfiguredException if a named url is used more than once in a menu?

**Comment by Gooliver2** [comment]
@mmarcos Added tests. 
About the `ImproperlyConfigured` exception: I didn't found nice way how to check it without too many if conditions and cycles. Current version is without hard limitation to number of params in `ARCTIC_MENU` rows (only title and url_name is required), so there is no other way than iter through  all lines, check types of params, recognize which one is the url_name and than check for uniqueness. 
`def menu()` method is triggered upon each request, so it will overload a bit overall request processing.
Or maybe you can give a tip how to validate it only once on starting Django app? 
Also now we have `max_weight` validation, so only submenu with the longest url will be displayed as active one. So, nothing bad will happen if settings will contains duplicates.
Maybe it will be better to add such checks after adding more strict rules for `ARCTIC_MENU`, with hard positioning of `('Title', 'url_name', 'icon', ((submenu,),), ('related_url))` items in settings.

File: arctic/utils.py (hunk 3 above)
Comment (Gooliver2) [reply arctic/utils.py]: added tests

**Comment by mmarcos** [comment]
Ok, let's leave it like this and if ever needed we can add extra checks.

**Pull request merged**